src/Sulu/Bundle/MediaBundle/Media/Storage/S3Storage.php

```diff
@@ -44,14 +46,16 @@ public function __construct(FilesystemInterface $filesystem, int $segments)
 
         $this->endpoint = (string) $this->adapter->getClient()->getEndpoint();
         $this->bucketName = $this->adapter->getBucket();
+
+        $this->url = $url;
```

Comment:
We can do the fallback already here:

```suggestion
        $this->url = null !== $url ? $url : $this->endpoint;
```